app/jobs/yoksis/prospective_students_save_job.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module Yoksis
+  class ProspectiveStudentsSaveJob < ApplicationJob
+    queue_as :high
+
+    def perform(type, year, academic_term: AcademicTerm.current)
+      @type          = type
+      @year          = year.to_i
+      @academic_term = academic_term
+      process
+    end
+
+    private
+
+    def process(page: 0)
+      response = Xokul::Yoksis::Prospectives.all(@type, @year, page: page) || {}
+
+      response.fetch(:data, []).each do |params|
+        Actions::ProspectiveStudent::Upsert.call(
+          params.merge(year: @year), academic_term: @academic_term
+        )
+      rescue ActiveRecord::RecordInvalid
+        next
```

Comment:
Loglayarak next yapsak?

Reply:
Hocam bunu loglayalım ama bakılmayacak logun(bu konu özelinde) çokta bir manası yok. Benim tür entegrasyon hatalarında farklı bir log mekanizmasına ihtiyacımız olduğu kanaatindeyim. Loglar arayüzde görüntülenmeli ve  sorun oluşan kayıt için belki işlem tekrar edilebilmeli. Bunu kullanan kişinin hatalardan haberdar olması lazımki gerekli durumunda düzeltmeyi vs yapabilisin.

Reply:
> Bunu kullanan kişinin hatalardan haberdar olması lazımki gerekli durumunda düzeltmeyi vs yapabilisin.

Evet.

Hataları sessiz şekilde geçmek doğru değil.  Yapılabilecek en minimal hareket loglamaktır, bu nedenle loglayalım dedim.  Bunun "optimal"i nedir, her durum için senin dediğin yönde tayin etmemiz lazım.  Ama optimali gelinceye kadar minimalle gidelim.

Reply:
Kodun her tafafına bu açıdan bakmak lazım aslında.  Ben gördükçe not alıyorum ama eminim ki böyle başka "case"ler de var.  Şuna ne dersiniz?  Tespit ettiğimiz tüm durumları "prensip olarak" loglayalım, logladığımız satırın yanına da `FIXME` veya `TODO` gibi comment düşelim.  Optimal çözümü bulduğumuzda bu commentlere döneriz.

Reply:
En azindan `Rollbar.info('Foo bar', response: response)` yapilabilir.